dashboard/src/components/settings/secrets/BaseSecretForm/BaseSecretForm.tsx

```diff
@@ -0,0 +1,136 @@
+import { useDialog } from '@/components/common/DialogProvider';
+import Form from '@/components/common/Form';
+import Button from '@/ui/v2/Button';
+import Input from '@/ui/v2/Input';
+import { useEffect } from 'react';
+import { useFormContext } from 'react-hook-form';
+import * as Yup from 'yup';
+
+export interface BaseSecretFormValues {
+  /**
+   * Identifier of the environment variable.
+   */
+  id: string;
+  /**
+   * The name of the role.
+   */
+  name: string;
+  /**
+   * Value of the secret.
+   */
+  value: string;
+}
+
+export interface BaseSecretFormProps {
+  /**
+   * Determines the mode of the form.
+   *
+   * @default 'edit'
+   */
+  mode?: 'edit' | 'create';
+  /**
+   * Function to be called when the form is submitted.
+   */
+  onSubmit: (values: BaseSecretFormValues) => void;
+  /**
+   * Function to be called when the operation is cancelled.
+   */
+  onCancel?: VoidFunction;
+  /**
+   * Submit button text.
+   *
+   * @default 'Save'
+   */
+  submitButtonText?: string;
+}
+
+export const BaseSecretFormValidationSchema = Yup.object({
+  name: Yup.string()
+    .required('This field is required.')
+    .test('isEnvVarValid', `The name must start with a letter.`, (value) =>
+      /^[a-zA-Z]{1,}[a-zA-Z0-9_]*$/i.test(value),
+    ),
```

Comment:
This tests the first char but also subsequent chars.

If I enter `abba?!$%^&*()` i (think) would see the error message "The name must start with a letter.".

Reply:
Yeah, we need to change this in the env var code as well. I just copied it from there.